### PR TITLE
feat(): npm access

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/api/auth.ts [hunk 1]
```diff
@@ -552,3 +573,234 @@ export function userHasWritePermissions(
...
 }
+
+export function lsPackages(pattern: string, auth: any): Promise<any> {
+  return new Promise((resolve, reject) => {
+    let user = auth.users.find(u => u.name === pattern);
+    let result = {};
+    if (user) {
+      // user is owner
+      let packages = auth.packages.filter(p => p.owners.findIndex(o => o === user.name) !== -1)
+      .map(p => p.name);
```
Comment (jkuri) [inline src/api/auth.ts]: indentation.

File: src/api/auth.ts [hunk 2]
```diff
@@ -552,3 +573,234 @@ export function userHasWritePermissions(
...
+          resolve(result);
+        } else {
+          reject({ errorCode: 412,
+            errorMessage: `Error: Organization ${splitPattern[0]} does not exists` });
+        }
+      } else {
+        // organization is owner
+        if (auth.organizations.findIndex(o => o.name === pattern) !== -1) {
+          let packages = auth.packages.filter(p => p.organization === pattern)
+          .map(p => p.name);
```
Comment (jkuri) [inline src/api/auth.ts]: here too.

**Review by jkuri — APPROVED** [review]

File: src/api/auth.ts (hunk 1 above)
Comment (Izak88) [reply src/api/auth.ts]: done

File: src/api/auth.ts (hunk 2 above)
Comment (Izak88) [reply src/api/auth.ts]: done

**Pull request merged**